Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/HDRColorGrading.azsl

```diff
@@ -0,0 +1,212 @@
+    float m_colorGradingPreSaturation;
+    float m_colorFilterIntensity;
+    float m_colorFilterMultiply;
+    float m_kelvin;
+    float m_kelvinLumPreservation;
+    float m_kelvinColorMix;
+    float m_whiteBalance;
+    float m_whiteBalanceTint;
+    float m_splitToneBalance;
+    float m_splitToneMix;
+    float m_colorGradingPostSaturation;
+    float m_smhShadowsStart;
+    float m_smhShadowsEnd;
+    float m_smhHighlightsStart;
+    float m_smhHighlightsEnd;
+    float m_smhMix;
+
+    float3 m_channelMixingRed;
+    float3 m_channelMixingGreen;
+    float3 m_channelMixingBlue;
+
+    float4 m_colorFilterSwatch;
+    float4 m_splitToneShadowsColor;
+    float4 m_splitToneHighlightsColor;
+
+    float4 m_smhShadowsColor;
+    float4 m_smhMidtonesColor;
+    float4 m_smhHighlightsColor;
+
+    // my color grading output
+    float4 m_color;
+}
+
+// Below are the color grading functions. These expect the frame color to be in ACEScg space.
+// If a color parameter is introduced from external sources (e.g. Editor color picker which uses Linear RGB), 
+// it will be converted into ACEScg as needed.
+float3 ColorGradePostExposure (float3 frame_color, float exposure) 
+{
+    frame_color *= pow(2.0f, exposure);
+	return frame_color;
+}
+
+// The contrast equation is performed in ACEScc (logarithmic) color space.
+float3 ColorGradingContrast (float3 frame_color, float midgrey, float amount) 
+{
+    float contrast_adjustment = amount * 0.01f + 1.0f;
+    frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScg, ColorSpaceId::ACEScc);
+    frame_color = (frame_color - midgrey) * contrast_adjustment + midgrey;
+	return frame_color = TransformColor(frame_color.rgb, ColorSpaceId::ACEScc, ColorSpaceId::ACEScg);
+}
+
+// The swatch_color param expects a linear RGB value.
+float3 ColorGradeColorFilter (float3 frame_color, float3 swatch_color, float alpha) 
+{
+    swatch_color = TransformColor(frame_color.rgb, ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    swatch_color *= pow(2.0f, PassSrg::m_colorFilterIntensity);
+    float3 frame_adjust = frame_color * swatch_color;
+    return frame_color = lerp(frame_color, frame_adjust, alpha);
+}
+
+float3 ColorGradeHueShift (float3 frame_color, float amount) 
+{
+	float3 frame_hsv = TransformColor(frame_color, ColorSpaceId::ACEScg, ColorSpaceId::HSV);
+	float hue = frame_hsv.x + amount;
+	frame_hsv.x = RotateHue(hue, 0.0, 1.0);
+	return TransformColor(frame_hsv, ColorSpaceId::HSV, ColorSpaceId::ACEScg);
+}
+
+float3 ColorGradeSaturation (float3 frame_color, float control) 
+{
+	float v_luminance = CalculateLuminance(frame_color, ColorSpaceId::ACEScg);
+	return (frame_color - v_luminance) * control + v_luminance;
+}
+
+float3 ColorGradeKelvinColorTemp(float3 frame_color, float kTemp, float mix, float lumPreservation)
+{
+    float3 kColor    = TransformColor(KelvinToRGB(kTemp), ColorSpaceId::LinearSRGB, ColorSpaceId::ACEScg);
+    float3 oLum      = RGBToHSL(frame_color.rgb);
```

Comment:
The HSL transforms are a bit suspect here, all you're really trying to do is preserve luminance right? why not just pre-multiply the kelvin color by the luminance of the frame color?

Reply:
Do you mean there's no need to convert into HSL, and just do:
```
return frameColor.rgb * kColor.rgb * luminance
```
I'm not sure if that would yield the same result.

Reply:
It's likely different than what you have but probably more perceptually accurate since luminance takes into consideration the various perceived intensities or r, g, b.